bundles/org.openhab.binding.playstation/src/main/java/org/openhab/binding/playstation/internal/discovery/PlayStationDiscovery.java

```diff
@@ -0,0 +1,548 @@
+                } catch (SocketTimeoutException e) {
+                    break; // leave the endless loop
+                }
+            }
+        } catch (IOException e) {
+            logger.debug("No PS4 device found. Diagnostic: {}", e.getMessage());
+        }
+    }
+
+    private synchronized void discoverPS3() {
+        logger.debug("Trying to discover all PS3 devices that have \"Connect PS Vita System Using Network\" on.");
+
+        InetAddress bcAddress = getBroadcastAdress();
+        InetAddress localAddress = getIPv4Adress();
+
+        if (localAddress == null || bcAddress == null) {
+            logger.info("No IP/Broadcast address found. Make sure OpenHab is configured!");
+            return;
+        }
+        try (DatagramSocket socket = new DatagramSocket(0, getIPv4Adress())) {
+            socket.setBroadcast(true);
+            socket.setSoTimeout(DISCOVERY_TIMEOUT_SECONDS * 1000);
+
+            NetworkInterface nic = NetworkInterface.getByInetAddress(localAddress);
+            byte[] macAdr = nic.getHardwareAddress();
+            StringBuilder macBuilder = new StringBuilder();
+            for (int macIndex = 0; macIndex < macAdr.length; macIndex++) {
+                macBuilder.append(String.format("%02x", macAdr[macIndex]));
+            }
+            String macString = macBuilder.toString();
+            // send discover
+            StringBuilder srchBuilder = new StringBuilder("SRCH3 * HTTP/1.1\n");
+            srchBuilder.append("device-id:");
+            srchBuilder.append(macString);
+            srchBuilder.append("01010101010101010101\n");
+            srchBuilder.append("device-type:PS Vita\n");
+            srchBuilder.append("device-class:0\n");
+            srchBuilder.append("device-mac-address:");
+            srchBuilder.append(macString);
+            srchBuilder.append("\n");
+            srchBuilder.append("device-wireless-protocol-version:01000000\n\n");
+            byte[] discover = srchBuilder.toString().getBytes();
+            DatagramPacket packet = new DatagramPacket(discover, discover.length, bcAddress,
+                    DEFAULT_PS3_MEDIA_MANAGER_PORT);
+            socket.send(packet);
+
+            // wait for responses
+            while (true) {
+                byte[] rxbuf = new byte[512];
+                packet = new DatagramPacket(rxbuf, rxbuf.length);
+                try {
+                    socket.receive(packet);
+                    parsePS3Packet(packet);
+                } catch (SocketTimeoutException e) {
+                    break; // leave the endless loop
+                }
+            }
+        } catch (IOException e) {
+            logger.debug("No PS3 device found. Diagnostic: {}", e.getMessage());
+        }
+    }
+
+    /**
+     * This method is used to find all PS3 devices that are in Remote-Play mode.
+     * Unused as of now.
+     */
+    @SuppressWarnings("unused")
+    private synchronized void discoverPS3Old() {
```

Comment:
Actually, dead code shouldn't be part of a binding. Can you save it and remove it for now?